### PR TITLE
release: sourcegraph@4.2.0

### PR DESCRIPTION
**Pull request by coury-clark**
This pull request is part of the Sourcegraph 4.2.0 release.

* [Release batch change](https://k8s.sgdev.org/organizations/sourcegraph/batch-changes/release-sourcegraph-4.2.0)
* [Tracking issue](https://github.com/sourcegraph/sourcegraph/issues/43423)

### Test plan

CI checks in this repository should pass, and a manual review should confirm if the generated changes are correct.

### CODE REVIEW

File: base/otel-collector/otel-agent.DaemonSet.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
     spec:
       containers:
         - name: otel-agent
-          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7d3094b05e4d9149b986d094d632989f8b0bbdde806b22bcc877d07c982ca01
+          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7289759d7dec743d9370ab4c38057c2751640b61a1251c8071544bf05308e37
```
Comment (coury-clark) [inline base/otel-collector/otel-agent.DaemonSet.yaml:29]: this seems wrong

File: base/otel-collector/otel-collector.Deployment.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ spec:
     spec:
       containers:
         - name: otel-collector
-          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7d3094b05e4d9149b986d094d632989f8b0bbdde806b22bcc877d07c982ca01
+          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7289759d7dec743d9370ab4c38057c2751640b61a1251c8071544bf05308e37
```
Comment (coury-clark) [inline base/otel-collector/otel-collector.Deployment.yaml:29]: same image / tag as above

File: base/otel-collector/otel-agent.DaemonSet.yaml [hunk 3]
```diff
@@ -26,7 +26,7 @@ spec:
     spec:
       containers:
         - name: otel-agent
-          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7d3094b05e4d9149b986d094d632989f8b0bbdde806b22bcc877d07c982ca01
+          image: index.docker.io/sourcegraph/opentelemetry-collector:4.2.0@sha256:4bc1df3d5bc681f2b16c39a66f3517b7493cf993046ac3036e781ad7c9c30a65
```
Comment (coury-clark) [inline base/otel-collector/otel-agent.DaemonSet.yaml:29]: please double check this image. I had to manually fix this from insiders

File: base/otel-collector/otel-collector.Deployment.yaml [hunk 4]
```diff
@@ -26,7 +26,7 @@ spec:
     spec:
       containers:
         - name: otel-collector
-          image: index.docker.io/sourcegraph/opentelemetry-collector:insiders@sha256:c7d3094b05e4d9149b986d094d632989f8b0bbdde806b22bcc877d07c982ca01
+          image: index.docker.io/sourcegraph/opentelemetry-collector:4.2.0@sha256:4bc1df3d5bc681f2b16c39a66f3517b7493cf993046ac3036e781ad7c9c30a65
```
Comment (coury-clark) [inline base/otel-collector/otel-collector.Deployment.yaml:29]: please double check this image. I had to manually fix this from insiders

**Review by BolajiOlajide — APPROVED** [review]

**Pull request merged**